src/super_gradients/training/utils/callbacks/callbacks.py

```diff
@@ -856,6 +856,22 @@ def _infer_global_step(self, context: PhaseContext, is_train_loader: bool):
             return total_steps_in_done + train_loader_length + context.batch_idx
 
 
+@register_callback(Callbacks.SLIDING_WINDOW_INFERENCE)
+class ChangeToSWI(Callback):
+    """
+    Performing single-scale sliding window during inference at the end of training on the validation set.
+    """
+
+    def on_validation_loader_start(self, context: PhaseContext) -> None:
+        if context.training_params.max_epochs - 1 == context.epoch:
+            unwrap_model(context.net).enable_swi()
+            context.valid_loader.dataset.transforms.transforms = []
```

Comment:
I feel that emptying transforms list in this way is VERY risky move. That may be solving immediate goal of adding SWI but if someone who is less familiar with implementation details runs into troubles because of this change - the odds they would curse us are pretty high.
Maybe even now we may have some incompatible scenarios where this would break (Like quantization). I really feel uneasy about this line. Is there really not other way of solving this?

Reply:
I agree. It was a quick fix due to time frame concerns.
@shaydeci how should I proceed?

Reply:
Missed this one, completely agree with @BloodAxe here.
How about mentioning explicitly the new transforms we switch to when switching to SWI? Is it always going to be the case that we just empty the transforms ?
It does not make sense to me that transforms that might have nothing to do with resolution should be dropped...

Any way, we should set the transforms taken out after the validation is completed.
I think an appropriate warning/message should be printed as well.

Reply:
If we drop transforms to keep spatial size, how do we handle normalization transforms then?

Reply:
I am not 100 percent sure, but think we need to call __iter__ otherwise this won't be updated. Have you checked that images actually pass through this empty pipeline ?

Reply:
This also makes sense to me as:

```
    def _transform_image_and_mask(self, image, mask) -> tuple:
        """
        :param image:           The input image
        :param mask:            The input mask
        :return:                The transformed image, mask
        """
        transformed = self.transforms({"image": image, "mask": mask})
```

This would'nt have worked...

Reply:
Fixed on 0ec8a9c